src/mms_client/utils/multipart_transport.py

```diff
@@ -0,0 +1,247 @@
+XOP = "http://www.w3.org/2004/08/xop/include"
+XMIME5 = "http://www.w3.org/2005/05/xmlmime"
+FILETAG = "xop:Include:"
+ID_LEN = 16
+
+
+# Define a function to generate a randomized ID string
+def get_id(length: int = ID_LEN) -> str:
+    """Generate a randomized ID string.
+
+    Arguments:
+    length (int):   The length of the ID string to generate.
+
+    Returns:    The randomized ID string.
+    """
+    return "".join(SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(length))
+
+
+def now_b64():
+    """Return a base64 encoded string of the current timestamp."""
+    return b64encode(f"{now().timestamp()}".replace(".", "").encode("UTF-8")).decode("UTF-8")
+
+
+def get_boundary() -> str:
+    """Return a randomized MIME boundary string."""
+    return f"MIMEBoundary_{now_b64()}".center(33, "=")
+
+
+def get_content_id() -> str:
+    """Return a randomized MIME content ID string."""
+    return f"<{now_b64()}@electroroute.co.jp>"
+
+
+def overwrie_attachnode(node):
+    """Overwrite the attachment node.
+
+    Arguments:
+    node (Element):    The XML node to be overwritten.
+
+    Returns:    The attachment node.
+    """
+    cid = node.text[len(FILETAG) :]
+    node.text = None
+    etree.SubElement(node, f"{{{XOP}}}Include", nsmap={"xop": XOP}, href=f"cid:{cid}")
+    return cid
+
+
+def get_multipart(content_id: str) -> MIMEMultipart:
+    """Create a MIME multipart object.
+
+    Arguments:
+    content_id (str):   The content ID to be used for the attachment.
+
+    Returns:    The MIME multipart object.
+    """
+    part = MIMEMultipart(
+        "related", charset="UTF-8", type="application/xop+xml", boundary=get_boundary(), start=content_id
+    )
+    part.set_param("start-info", "application/soap+xml")
+    return part
+
+
+def get_envelopepart(envelope: Element, content_id: str) -> MIMEApplication:
+    """Create the MIME envelope part.
+
+    Arguments:
+    envelope (Element):    The XML envelope to be attached.
+    content_id (str):      The content ID to be used for the attachment.
+
+    Returns:    The MIME envelope part.
+    """
+    part = MIMEApplication(etree_to_string(envelope), "xop+xml", encode_7or8bit)
+    part.set_param("charset", "utf-8")
+    part.set_param("type", "text/xml")
+    part.add_header("Content-ID", content_id)
+    part.add_header("Content-Transfer-Encoding", "binary")
+    return part
+
+
+class Attachment:
+    """Represents an attachment to be sent with a multipart request."""
+
+    def __init__(self, name: str, data: bytes):
+        """Create a new attachment.
+
+        Arguments:
+        name (str):   The name of the attachment.
+        data (bytes): The data to be attached.
+        """
+        self.name = name
+        self.data = data
+        self.cid = f"{get_id()}-{name}@electroroute.co.jp"
+        self.tag = FILETAG + self.cid
+
+
+class MultipartTransport(Transport):
+    """A transport that supports MTOMS attachments."""
```

Comment:
What is MTOMS? Is this a term that someone familiar with MMS would also know? Would this benefit from an explainer in the README?

Reply:
MTOM stands for "Message Transmission Optimization Mechanism" and it is used when sending XOP (XML-binary optimized protocol) messages. Basically, this is restricted to SOAP multi-part requests. Please forget everything you read now.